### PR TITLE
♻️ Remove PUT project endpoint

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
 - ♻️ Remove PUT project endpoint

## Related issue/s
- closes https://github.com/ITISFoundation/osparc-simcore/issues/6236

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by odeimaiz — COMMENTED** [review]
Long live the PUT

**Review by sanderegg — APPROVED** [review]
not sure I understand why we keep it? there is no one else but the frontend using that entrypoint right? why keep it at all?

File: api/specs/web-server/_projects_crud.py [hunk 1]
```diff
@@ -106,7 +106,8 @@ async def get_project(project_id: ProjectID):
 
 @router.put(
     "/projects/{project_id}",
-    response_model=Envelope[ProjectGet],
+    response_model=None,
+    status_code=status.HTTP_410_GONE,  # Deprecated endpoint
```
[the remark this answers is not in the document]
Comment (matusdrobuliak66) [reply api/specs/web-server/_projects_crud.py]: You are right, I have removed the PUT endpoint.

File: services/web/server/tests/unit/with_dbs/03/meta_modeling/test_meta_modeling_iterations.py [hunk 2]
```diff
@@ -144,13 +145,14 @@ async def test_iterators_workflow(
     project_data.update({key: modifications[key] for key in ("workbench", "ui")})
     project_data["ui"].setdefault("currentNodeId", project_uuid)
 
-    response = await client.put(
-        f"/v0/projects/{project_data['uuid']}",
-        json=project_data,
+    db: ProjectDBAPI = ProjectDBAPI.get_from_app_context(client.app)
```
[the remark this answers is not in the document]
Comment (matusdrobuliak66) [reply services/web/server/tests/unit/with_dbs/03/meta_modeling/test_meta_modeling_iterations.py:148]: It's fine, I have made the test work.

**Comment by matusdrobuliak66** [comment]
> not sure I understand why we keep it? there is no one else but the frontend using that entrypoint right? why keep it at all?

UPDATE: I have removed the PUT endpoint :+1:

**Review by giancarloromeo — APPROVED** [review]

**Review by pcrespov — APPROVED** [review]

**Pull request merged**